WordPress/gutenberg
# WIP Archive title: Display the correct title in the editor

## What?

Partial for https://github.com/WordPress/gutenberg/issues/44158

## Why?


## How?


## Testing Instructions
First, please register a custom post type and custom taxonomy. Remember to set show_in_rest to true.  I tested with this plugin which has both: https://wordpress.org/plugins/yoast-test-helper/

Make sure your WordPress install has at least one of each: custom post type, custom taxonomy, a published single post, a category, a tag.

In the Site Editor, create two archives of all possible types: One for "all items" and one for a specific item.
So your template list would include both "Author" and "Author: name here",  "Category", "Category: name here" and so on.

Make sure the templates have a query title block and that it displays correctly.
Disable the block option "Show archive type in title", check that the data still displays correctly.

## Screenshots or screencast
